plat/arm/common/arm_common.c

```diff
@@ -28,61 +28,34 @@ extern const mmap_region_t plat_arm_mmap[];
 
 /*
  * Set up the page tables for the generic and platform-specific memory regions.
- * The extents of the generic memory regions are specified by the function
- * arguments and consist of:
- * - Trusted SRAM seen by the BL image;
+ * The size of the Trusted SRAM seen by the BL image must be specified as well
+ * as an array specifying the generic memory regions which can be;
  * - Code section;
  * - Read-only data section;
  * - Coherent memory region, if applicable.
  */
-void arm_setup_page_tables(uintptr_t total_base,
-			   size_t total_size,
-			   uintptr_t code_start,
-			   uintptr_t code_limit,
-			   uintptr_t rodata_start,
-			   uintptr_t rodata_limit
-#if USE_COHERENT_MEM
-			   ,
-			   uintptr_t coh_start,
-			   uintptr_t coh_limit
-#endif
-			   )
+
+void arm_setup_page_tables(const mmap_region_t bl_regions[],
+			   const mmap_region_t plat_regions[])
 {
+#if LOG_LEVEL >= LOG_LEVEL_VERBOSE
+	const mmap_region_t *regions = bl_regions;
+
+	while (regions->size) {
+		VERBOSE("Region: %p - %p has attributes 0x%x\n",
+				(void *) regions->base_va,
+				(void *) (regions->base_va + regions->size),
+				regions->attr);
+		regions++;
+	}
+#endif
 	/*
 	 * Map the Trusted SRAM with appropriate memory attributes.
 	 * Subsequent mappings will adjust the attributes for specific regions.
 	 */
-	VERBOSE("Trusted SRAM seen by this BL image: %p - %p\n",
```

Comment:
We lose these printouts which can be useful for debug. Dropping these really seems to be the only functional thing this patch does.

Reply:
In a couple of the other tasks we are working on we have found the need to be able to map a varying number of memory regions. The old implementation of the function would require variable argument list to the function which we decided was not a clean way of solving the problem. This does however mean we lose some specific information such as the name of the region however I tried to reduce the impact of this by printing out the attributes which can be used to work out the regions being mapped.

Reply:
What tasks, variable number of memory regions are supported by the second parameter.

Reply:
One of the tasks I'm working on is to reclaim initialisation code in BL31 for secondary CPU stack usage. To this I need to move the init code section to below the data section. We need to map a region for this section however this cannot be added to the array in second parameter as the location of the section is only known after linking.

Reply:
I am also working in a task to share part of the libraries between different boot loaders. It means that 2 additional regions can be needed.